packages/next-app/src/components/Link.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import { Url } from "next/dist/shared/lib/router/router";
+import Link from "next/link";
+import React from "react";
+
+import { Spinner, SpinnerColor } from "./Spinner";
+
+export function LinkComponent({
```

Comment:
Cool! Would be helpful to see a demo of this to review

Reply:
Sure! Now that the machine on fly is back to normal this will be probably less needed but still a great plan B to make sure the UX is great

https://www.loom.com/share/bec9ff4be2c74c61bb2334123414e9f1?sid=712e00da-6bbb-41bf-b645-a3a9a14efcaf